subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/problems/ConfigurationCacheReport.kt

```diff
@@ -132,14 +135,18 @@ class ConfigurationCacheReport(
                 return this
             }
 
-            override fun commitReportTo(outputDirectory: File, cacheAction: String, requestedTasks: String, totalProblemCount: Int): Pair<State, File?> {
-                lateinit var reportFile: File
+            override fun commitReportTo(outputDirectory: File, cacheAction: String, requestedTasks: String, totalProblemCount: Int, taskExecutionProblems: Int): Pair<State, File?> {
+                val reportFile: AtomicReference<File> = AtomicReference(null)
```

Comment:
I was under the impression the compiler would generate thread-safe code for a `lateinit` var but looking at the bytecode now I can see it's not the case. Good move 👍

Reply:
Ended up taking that bit back, as changes to CCR were no longer needed

Reply:
(BTW, the reason for this change is that lateinit vars cannot be nullable)

Reply:
Got it.